### PR TITLE
add exit code on verification failure

### PR DESCRIPTION
**Pull request by vgropp**
I was in need for an exit code on verification failure to handle it in some script.

What do you think?

### CODE REVIEW

**Comment by Wunkolo** [comment]
This is a good feature, so that a failed check or error can be communicated to outer automation-scripts and such.

File: source/main.cpp [hunk 1]
```diff
@@ -330,11 +330,12 @@ int Check(const Settings& CurSettings)
 	}
 
 	std::vector<std::thread> Workers;
+	int ExitCode { EXIT_SUCCESS };
```
Comment (Wunkolo) [inline source/main.cpp]: Though the proposed code works, it feels like a bit of a bad pattern to rely on multiple threads to over-write an `int` without any synchronization promises, even if the code can be pretty easily reasoned with. Maybe an `std::atomic<bool> Error{false};` with each thread calling `Error.fetch_or(!Valid)` or `Error.fetch_or(true)` would be better-utilized here to guarantee things are coherent.

Comment (Wunkolo) [reply source/main.cpp]: Finding out that `std::atomic<bool>` does not implement `fetch_or`.
Maybe just `std::atomic<std::size_t> Passed{0};` and calling `fetch_add` to count up successful checks and returning `EXIT_SUCCESS`/`EXIT_FAILURE` depending on if _all_ of them passed?

**Comment by vgropp** [comment]
anything you want me to add before merging?

**Review by Wunkolo — CHANGES_REQUESTED** [review]
Ah I realized my reviews were "pending" and not visible to you this entire time.

File: source/main.cpp (hunk 1 above)
Comment (vgropp) [reply source/main.cpp]: I understand your point, will look into it the next days

Comment (vgropp) [reply source/main.cpp]: changed to a passed atomic. Really like it more, cause its only adding/changing for passes and comparing those for the exit value

**Pull request merged**